src/main/java/org/springframework/integration/dsl/IntegrationFlowDefinition.java

```diff
@@ -2346,7 +2347,7 @@ public void accept(GenericEndpointSpec<BridgeHandler> bridge) {
 			}
 		}
 		if (hasSubFlows) {
-			channel(afterRouterChannel);
+			handle(bridgeHandler);
```

Comment:
It's a bit tricky to predict the next `IntegrationFlow` step at this point.
As you see I had a `DirectChannel` before...

IMO `BridgeHandler` is better than channel shifting.

So, continuing with the flow we should have the channel here, but allow to rely on the `replyChannel` it would be better to avoid extra channel.
Having that the `BridgeHandler` is the best choice: it doesn't modify message and we can avoid some undesired global channel interceptor.